src/shogun/io/TBOutputFormat.cpp

```diff
@@ -106,6 +107,13 @@ tensorflow::Event TBOutputFormat::convert_scalar(
 		    "Unsupported type %s", value.first.get_value().type_info().name());
 	}
 
+	auto write_summary = [&](auto type) {
```

Comment:
Question, could we explicit here with the variables passed on as reference?

Reply:
Do you mean something like:
```
auto write_summary = [value](auto type) {summaryValue->set_simple_value(
 		    any_cast<decltype(type)>(value.first.get_value()));
 	};
```

Reply:
like that yes, so that callers are forced to think about the scope

Reply:
ok great, `this` is quite redundant anyways, I like the C++ updates :)

Reply:
Yup, now just need to come up with a way to make sure people use this signature... 
Would it be worth having some checks (not sure how)? Or just add somewhere in the documentation?

Reply:
I think it is fine to let people look at examples and then use the same style. Documentation just outdates, and this is only for developers (who know how to browse the codebase) anyways.
What happens if a different signature is passed. Compile error I guess so that would be fine.